addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/RequestorJob.java

```diff
@@ -0,0 +1,182 @@
+                                        request));
+                    }
+                    String responseCode = (String) requestMap.get("responseCode");
+                    if (responseCode != null) {
+                        try {
+                            int code = Integer.parseInt(responseCode);
+                            if (code < 100 || code > 599) {
+                                progress.warn(
+                                        Constant.messages.getString(
+                                                "automation.error.requestor.badcode",
+                                                this.getName(),
+                                                responseCode,
+                                                request));
+                            }
+                        } catch (NumberFormatException e) {
+                            progress.warn(
+                                    Constant.messages.getString(
+                                            "automation.error.requestor.badcode",
+                                            this.getName(),
+                                            responseCode,
+                                            request));
+                        }
+                    }
+                }
+            }
+        } else {
+            progress.error(Constant.messages.getString("automation.error.requestor.badlist", o));
+        }
+    }
+
+    @Override
+    public void runJob(
+            AutomationEnvironment env, LinkedHashMap<?, ?> jobData, AutomationProgress progress) {
+        ArrayList<?> requests = (ArrayList<?>) jobData.get("requests");
+        for (Object request : requests) {
+            LinkedHashMap<?, ?> requestMap = (LinkedHashMap<?, ?>) request;
+            HttpMessage msg = new HttpMessage();
+            msg.getRequestHeader().setMethod((String) requestMap.get("method"));
+            try {
+                msg.getRequestHeader().setURI(new URI((String) requestMap.get("url"), true));
+            } catch (URIException e) {
+                // Handled above
+            }
+            String name;
+            if (requestMap.get("name") != null) {
+                name = (String) requestMap.get("name");
+            } else {
+                name =
+                        msg.getRequestHeader().getMethod()
+                                + msg.getRequestHeader().getURI().toString();
+            }
+            if (requestMap.get("data") != null) {
+                msg.getRequestBody().setBody((String) requestMap.get("data"));
+            }
+            try {
+                httpSender.sendAndReceive(msg);
+            } catch (IOException e) {
+                progress.warn(
+                        Constant.messages.getString(
+                                "automation.error.requestor.badnetwork", this.getName(), name, e));
+            }
+            if (requestMap.get("responseCode") != null) {
```

Comment:
Could store it in an int so you dont have to get it again, or just check that the map contains the key

Reply:
Oh and dont forget that the user can ignore warnings, in which case the responseCode might not be an int ;)